src/spec-configuration/containerFeaturesOCI.ts

```diff
@@ -15,7 +15,7 @@ export function tryGetOCIFeatureSet(output: Log, identifier: string, options: bo
 		value: options
 	};
 
-	const userFeatureIdWithoutVersion = originalUserFeatureId.split(':')[0];
+	const userFeatureIdWithoutVersion = originalUserFeatureId.split(':').slice(0, -1).join(':');
```

Comment:
For OCI-based Features this could be changed to `getRef(...).resource` - that code is where i've been centralizing the parsing of an OCI Feature identifier, and should handle this case.

Reply:
I'd be curious if this property could be removed entirely, and change all instances to use the `featureRef`?

Reply:
Actually, it looks like I may be able to remove the `userFeatureIdWithoutVersion` entirely with the in-flight dependency work (experimental implementation here: https://github.com/devcontainers/cli/pull/511)

Reply:
I've made it use `getRef(...).resource` for OCI-based features and removed the duplicate code where possible by using `sourceInformation.userFeatureIdWithoutVersion || sourceInformation.userFeatureId`. It looks quite clean now